### PR TITLE
Fix regression where E2ETest app no longer uploads tree dump outputs

### LINKED ISSUE #6684: CI Loop:  tree dump comparison file not being uploaded as artifacts
@chiaramooney  ran into this issue while working on:
https://github.com/microsoft/react-native-windows/pull/6609

The problem we're seeing is that when one of the E2E tests fails a master treedump comparison, the comparison file is not being uploaded as a test artifact.  This makes it impossible to see what actually failed by looking at the logs.

I created a little PR to repro this issue here:
https://github.com/microsoft/react-native-windows/pull/6683
Go to the CI loop ADO page and dig into the E2E test failure.  Notice you cannot locate the test comparison file, which is supposed to be here:
e2e-errors

Chiara and I debugged together and can see that when we run the test locally, the expected comparison file is generated on the local box.  This makes me think the issue is somewhere in the artifact publishing process.

### PR DESCRIPTION
**Pull request by asklar**
Fixes #6684 

As part of #6318, there was an issue where the last visited page would be brought up on app launch. To fix that, the application data was cleared which introduced a regression since the tree dump output was being saved to the app's local folder

The fix, since I didn't want to re-introduce the last-visited-page regression, is to store the tree dump in the Documents folder instead of appdata. This requires adding a new capability to the uwp app and specifying the file types we save.

Separately, during #6125, the publisher name was updated but the certificate wasn't re-created. As a result we were left with a package family name that didn't match the publisher name, which broke the assumptions on where the app would get deployed.
This resulted in a build time warning that was easily missed (package publisher does not match certificate file or something like that). As a result, the tree dump files could not be found at the expected location and would therefore not be published.

Sample run where it fails (correctly now): https://dev.azure.com/ms/react-native-windows/_build/results?buildId=127909&view=results

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6686)

### CODE REVIEW

File: .ado/jobs/e2e-test.yml [hunk 1]
```diff
@@ -101,12 +101,62 @@ jobs:
...
+          script: type C:\Users\VssAdministrator\AppData\Local\Packages\ReactUWPTestApp_cezq6h4ygq1hw\LocalState\ControlStyleRegularBorder.json
+          workingDirectory:  $(LocalAppData)\Packages
+        condition: succeededOrFailed()
+
+
       - task: CopyFiles@2
         displayName: Copy tree dump output files
         inputs:
-          sourceFolder: $(LocalAppData)\Packages\ReactUWPTestApp_kc2bncckyf4ap\LocalState
+          sourceFolder: $(LocalAppData)\Packages\ReactUWPTestApp_cezq6h4ygq1hw\LocalState
```
Comment (dannyvv) [inline .ado/jobs/e2e-test.yml:157]: Should we create a pipeline variable/parameter for this value?

**Review by dannyvv — APPROVED** [review]
Wow, this dependency is super tricky....

File: .ado/jobs/e2e-test.yml (hunk 1 above)
Comment (asklar) [reply .ado/jobs/e2e-test.yml:157]: makes sense - I'm still waiting to hear back from virtual environments team on what I'm seeing where the app sees the files being created, but after the app exits, the files are not there: filed https://github.com/actions/virtual-environments/issues/2207

**Comment by kmelmon** [comment]
Thanks @asklar for fixing this!

**Pull request merged**